blocks/library/code-mirror/index.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import CodeMirror from 'react-codemirror';
+import he from 'he';
+require( 'codemirror/mode/css/css' );
+require( 'codemirror/mode/diff/diff' );
+require( 'codemirror/mode/javascript/javascript' );
+require( 'codemirror/mode/markdown/markdown' );
+require( 'codemirror/mode/pegjs/pegjs' );
+require( 'codemirror/mode/php/php' );
+require( 'codemirror/lib/codemirror.css' );
+
+/**
+ * Internal dependencies
+ */
+import { registerBlockType, query } from '../../api';
+const { html } = query;
+
+const toEditor = s => he.decode( s.split( '<br>' ).join( '\n' ) );
+const fromEditor = s => he.encode( s, { useNamedReferences: true } ).split( '\n' ).join( '<br>' );
```

Comment:
thanks! I pulled it in initially because it was a prototype. I'm very glad you were aware of how `<pre><code>` changed the semantics. much better now!